Lib/trace.py

```
@@ -167,12 +167,12 @@ def __init__(self, counts=None, calledfuncs=None, infile=None,
         self.callers = self.callers.copy()
         self.infile = infile
         self.outfile = outfile
-        if self.infile:
+        if self.infile is not None:
```

The effect of this change is that infile=='' will result in open raising an OSError subclass instead of silently doing nothing.  If someone wants the latter, they should pass None.